dGame/Entity.cpp

```diff
@@ -946,12 +926,17 @@ void Entity::WriteBaseReplicaData(RakNet::BitStream* outBitStream, eReplicaPacke
 			outBitStream->Write0(); //No ldf data
 		}
 
-		if (m_Trigger != nullptr && m_Trigger->events.size() > 0) {
-			outBitStream->Write1();
-		} else {
+		TriggerComponent* triggerComponent;
+		if (TryGetComponent(COMPONENT_TYPE_TRIGGER, triggerComponent)) {
+			// has trigger component, check to see if we have events to handle
+			auto* trigger = triggerComponent->GetTrigger();
+			if (trigger && trigger->events.size() > 0) outBitStream->Write1();
+			else outBitStream->Write0();
+		} else { // no trigger componenet, so definately no triggers
```

Comment:
We should check client logs here to verify there are no changes in behavior here.
Since this is a hot code path, I am going to nit about the speed:
An if:else is a bit overkill and adds needless branching and we could instead just do `Write<bool>(trigger && trigger->events.size() > 0);`